tests/testBootprint.py

```diff
@@ -0,0 +1,246 @@
+        self.schema = afwDet.PeakTable.makeMinimalSchema()
+        # Run the the constructors test to ensure the Bootprints are setUp
+        self.testConstructors()
+
+    def tearDown(self):
+        del self.spans
+        del self.region
+        del self.schema
+        del self.footprint
+        del self.footprintWithRegion
+        del self.footprintWithSchema
+        del self.footprintWithSchemaRegion
+
+    def testConstructors(self):
+        '''
+        Test that each of the constructors constructs a valid Bootprint,
+        if any of these fails, an exception will be raised and the test
+        will fail.
+        '''
+        self.footprint = afwDet.Bootprint(self.spans)
+        self.footprintWithRegion = afwDet.Bootprint(self.spans, self.region)
+        self.footprintWithSchema = afwDet.Bootprint(self.spans, self.schema)
+        self.footprintWithSchemaRegion = afwDet.Bootprint(self.spans,
+                                                          self.schema,
+                                                          self.region)
+
+    def testIsHeavy(self):
+        self.assertFalse(self.footprint.isHeavy)
+
+    def testGetSetSpans(self):
+        '''
+        Test getting and setting the SpanSet member of the Bootprint with both
+        the getters and setters and the python property accessor
+        '''
+        self.assertTrue(self.footprint.getSpans() == self.spans)
+        self.assertTrue(self.footprint.spans == self.spans)
+        tempSpanSet = afwGeom.SpanSet.spanSetFromShape(2, afwGeom.Stencil.BOX)
+        self.footprint.setSpans(tempSpanSet)
+        self.assertTrue(self.footprint.spans == tempSpanSet)
+        # reset back to origional with property
+        self.footprint.spans = self.spans
+        self.assertTrue(self.footprint.spans == self.spans)
+
+    def testPeakFunctionality(self):
+        newSchema = afwDet.PeakTable.makeMinimalSchema()
+        newField = afwTable.Field_I("extra", "doc", "na")
+        newSchema.addField(newField)
+        self.footprint.setPeakSchema(newSchema)
+        names = self.footprint.getPeaks().getSchema().getNames()
+        self.assertTrue("extra" in names)
+        # reset the schema back
+        self.footprint.setPeakSchema(self.schema)
+        peakData = [[2, 2, 10], [0, 3, 21], [1, 9, 17]]
+        for peak in peakData:
+            self.footprint.addPeak(*peak)
+        # Sort the peaks by value (use the property peaks to test that method
+        # of access)
+        sortKey = self.footprint.peaks.getSchema()['peakValue'].asKey()
+        self.footprint.sortPeaks(sortKey)
+        for i, peak in enumerate(self.footprint.peaks):
+            self.assertEqual(peak['i_x'], i)
+
+        # Test that peaks outside the Bootprint are removed
+        self.footprint.removeOrphanPeaks()
+        self.assertEqual(len(self.footprint.peaks), 2)
+        for peak in self.footprint.peaks:
+            self.assertNotEqual(peak['i_x'], 1)
```

Comment:
a new one is constructed at the beginning of each test function by the setUp function